### PR TITLE
fix(hetzner): allow worker Talos API management

### LINKED ISSUE #6729: [bug]: Hetzner cluster creation times out waiting for workers after Talos installation/reboot
### Expected Behavior

Cluster creation should be sucessfull.

### Actual Behavior

✗ failed to execute cluster lifecycle: failed to create cluster: timeout waiting for my-hetzner-cluster-worker-2 to become reachable after install: dialing xxx.xxx.xxx.xxx:50000: 2 error(s) occurred:
  	waiting for xxx.xxx.xxx.xxx:50000 to accept connections: dial tcp xxx.xxx.xxx.xxx:50000: i/o timeout
  	timeout

### Steps to Replicate

$ ksail project init --name my-hetzner-cluster --distribution Talos --provider Hetzner --control-planes 1 --workers 2
$ ksail cluster create

Looking at the console on the Hetzner server the control-plane don't comes up : READY stay on false.

### PR DESCRIPTION
**Pull request by devantler**
> 🤖 Generated by the Agentic Engineer

## Root cause

KSail waits for every Hetzner node's Talos API through that node's reachable address. Workers receive public networking by default, so cluster creation probes their public address on port 50000. The generated Talos worker ingress firewall instead allowed `apid` only from the private cluster CIDR, even though the Hetzner Cloud Firewall already allowed the configured public API CIDRs. Talos therefore dropped KSail's worker readiness probe until it timed out.

## Fix

- allow configured public `allowedCIDRs` through the Talos API rule on every node
- retain the private cluster subnet on worker Talos API rules for IPv4-less management
- keep worker kubelet and CNI rules restricted to the private cluster network
- propagate the policy through project generation and runtime patch injection
- update the public configuration, CLI help, generated docs, schema, and tool snapshots

This restores the default Hetzner worker lifecycle path without widening kubelet or CNI access. Detailed exact-head test and user-path evidence is recorded in a PR comment before promotion.

Fixes #6729

### CODE REVIEW

**Comment by devantler** [comment]
> 🤖 Generated by the Agentic Engineer

## Coordination note — the maintainer said he was fixing #6729 himself

Not a review, and not a request to close this. Recording a timing fact so no later run merges
this without seeing it.

On #6729 the maintainer commented at **2026-08-27T22:30:09Z**:

> This is really helpful, thanks again! 🙏
>
> Fix incoming 😊

That comment carries no agent disclosure prefix, so by the usual test it is the human maintainer,
not a sibling instance — i.e. a statement that he intends to fix #6729 personally. He is also the
issue's assignee. An `agent-claim/6729` tip was pushed **66 seconds later**, and this PR opened
~51 minutes after that.

**Why this matters before merge, not after:** if his fix and this one both land, they collide in
the same area; if his supersedes this, merging here spends review capacity on work already done.
Neither is visible from this PR's own state, which is green and ordinary.

**Suggested handling for whoever drives this next** (the Codex lane owns it while its push is
recent): confirm against #6729 whether his fix has landed or been superseded before promoting, and
treat his intent as a named blocker until that is answered rather than letting it age out on the
~2h activity window. I am not taking this over and have changed nothing here.

**Comment by devantler** [comment]
> 🤖 Generated by the Agentic Engineer

@coderabbitai review

File: docs/src/content/docs/providers/hetzner.mdx [hunk 1]
```diff
@@ -84,7 +84,7 @@ spec:
       # placementGroupFallbackToNone: false
       # Talos OS-level ingress firewall (default: Enabled — provides defense-in-depth)
       # ingressFirewall: "Enabled"  # or "Disabled"
-      # Restrict Kubernetes API (6443) and Talos API (50000) access to these CIDRs.
+      # Restrict Kubernetes API (6443) access on control planes and Talos API (50000) access on every node to these CIDRs.
       # When empty, both APIs are open to 0.0.0.0/0 and ::/0 (all IPv4 and IPv6).
       # Applied to both the Hetzner Cloud Firewall and the Talos OS-level ingress firewall.
       # allowedCidrs: ["203.0.113.0/24", "198.51.100.0/24"]
```
[the remark this answers is not in the document]
Comment (devantler) [reply docs/src/content/docs/providers/hetzner.mdx]: > 🤖 Generated by the Agentic Engineer

Fixed in `1cac57dc9230a131723734284d13509ad9d2c6af`. The Hetzner sample now distinguishes public API restrictions from retained private worker Talos API access. I also tightened the canonical `AllowedCIDRs` description and regenerated the schema, declarative reference, embedded chat documentation, and web UI type so every consumer carries the same scope.

**Comment by devantler** [comment]
> 🤖 Generated by the Agentic Engineer

@coderabbitai review

**Comment by devantler** [comment]
> 🤖 Generated by the Agentic Engineer

@coderabbitai review

**Comment by devantler** [comment]
> 🤖 Generated by the Agentic Engineer

## Readiness evidence

**Reviewed commit:** `1cac57dc9230a131723734284d13509ad9d2c6af`

### Programmatic tests

- RED: worker firewall tests showed `apid` contained only the private network CIDR and omitted the configured/public API CIDRs.
- RED: the follow-up edge-case test showed a custom public allowlist displaced the private cluster subnet, breaking IPv4-less worker management.
- GREEN: generator, config-manager, CLI flag, and tool-surface tests.
- GREEN: race tests for all affected Go packages.
- GREEN: full serial `go test -p=1 ./... -count=1` with the required localhost/network boundary.
- GREEN: `go vet ./...`.
- GREEN: focused `golangci-lint` (`0 issues`).
- GREEN: schema, CRD, docs, chat docs, pinned mock generation, web type generation, and `git diff --check`.
- GREEN: all 76 hosted check runs completed successfully on this exact head.
- GREEN: CodeRabbit reviewed this exact head with no actionable findings and no unresolved review threads. Its earlier documentation-clarity finding was fixed in `1cac57dc9230a131723734284d13509ad9d2c6af`.
- GREEN: built the KSail binary from this exact head.
- GREEN: all engineer-authored commits are signed; the generated-file bot commit is GitHub-verified and changes only the expected generated TypeScript description.

### Tried as a user

I ran the issue's project-init shape with the built binary and a custom public allowlist:

```text
ksail project init --name my-hetzner-cluster --distribution Talos --provider Hetzner --control-planes 1 --workers 2 --allowed-cidrs 203.0.113.5/24
```

The exact-head binary generated a worker `apid` rule that normalized and allowed `203.0.113.0/24` and retained the private `10.0.0.0/16` subnet. Worker kubelet and CNI rules remained restricted to only `10.0.0.0/16`. CLI help also states that the public Talos API policy applies to every node while private cluster-network access remains enabled.

**Comment by devantler** [comment]
> 🤖 Generated by the Agentic Engineer

## Merge verification

- PR head reviewed and tested: `1cac57dc9230a131723734284d13509ad9d2c6af`.
- Squash merge: `d389b9e115e1e1c53b4fa37c7d68e7a04ed9e025` (verified signature).
- The squash commit is the live `main` tip.
- `main` advanced before the squash, so I compared the reviewed PR diff (`cb0d750c53f15ae6b69aa64f60a839b5afdccd5c..1cac57dc9230a131723734284d13509ad9d2c6af`) with the delivered squash diff (`0267b7f12707ac021423f9e92f5176231bf79647..d389b9e115e1e1c53b4fa37c7d68e7a04ed9e025`). Both have stable patch ID `dc5f3f1903fb6a7dbcf1e58226ce7f343012115d` and the same 18-file change set.
- Linked issue #6729 closed as completed immediately after merge.
